R/data-types.R

```diff
@@ -0,0 +1,40 @@
+setGeneric("dbiDataType", function(x) standardGeneric("dbiDataType"))
+
+data_frame_data_type <- function(x) {
+  vapply(x, dbiDataType, FUN.VALUE = character(1), USE.NAMES = TRUE)
+}
+
+varchar_data_type <- function(x) {
+  paste0("VARCHAR(", max(nchar(as.character(x))), ")")
```

Comment:
It does not make sense to me to limit the string length here. This will break lots of time, e.g. when people make a table with a data frame and then try to add more data to it. Also, it really depends on the DB implementation whether STRING/TEXT columns are wasteful. In MonetDB it actually is more expensive to have a limit because it needs to be enforced.